File: inventory_manager_frontend/src/services/ProductService/ProductService.ts
import axios, { AxiosResponse } from "axios";
import { getAxiosConfig } from "../config";
import {ProductResponse, ListProductResponse, ProductFilterRequest, ProductRequest} from "./types";

class ProductService {
  static async filter(filter: ProductFilterRequest): Promise<AxiosResponse<ListProductResponse>> {
    return axios.get(`/products`, { ...getAxiosConfig(), params: filter });
  }

  static async create(
    product: ProductRequest
  ): Promise<AxiosResponse<ProductResponse>> {
    return axios.post(`/products`, {...product}, { ...getAxiosConfig() });
  }

  static async update(
    id: number,
    product: ProductRequest
  ): Promise<AxiosResponse<ProductResponse>> {
    return axios.put(`/products/${id}`, {...product}, { ...getAxiosConfig() });
  }

  static async getById(id: number): Promise<AxiosResponse<ProductResponse>> {
    return axios.get(`/products/${id}`, { ...getAxiosConfig() });
  }

  static async delete(id: number): Promise<AxiosResponse<ProductResponse>> {
    return axios.delete(`/products/${id}`, { ...getAxiosConfig() });
  }
}
export default ProductService;

File: inventory_manager_service/src/main/java/sapo/team3/inventory_manager_service/service/ProductService.java
package sapo.team3.inventory_manager_service.service;

import sapo.team3.inventory_manager_service.model.response.PagingListResponse;
import sapo.team3.inventory_manager_service.model.response.product.ProductFilterRequest;
import sapo.team3.inventory_manager_service.model.response.product.ProductRequest;
import sapo.team3.inventory_manager_service.model.response.product.ProductResponse;

public interface ProductService {
    ProductResponse getById(long id);

    ProductResponse create(ProductRequest input);

    ProductResponse update(long id, ProductRequest input);

    ProductResponse delete(long id);

    PagingListResponse<ProductResponse> filter(ProductFilterRequest filter);
}
